modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,68 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  it('TINY-8641: Dispatch change and set editor dirty if the current content and the undoManager history are incoherent', () => {
+    const editor = hook.editor();
+
+    let changeEventCounter = 0;
+    let currentChangeEvent: { level: string; lastLevel: string };
+
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevel = 'a modified last level';
+
+    const assertChangeEvent = (event: { level: string; lastLevel: string }) => {
+      assert.equal(event.level, initialContent, 'Level has not the expected content');
+      assert.equal(event.lastLevel, manualModifiedLevel, 'Last level has not the expected content');
+    };
+
+    editor.resetContent(initialContent);
+    const onChange = (e: EditorEvent<{
+      level: UndoLevel;
+      lastLevel: UndoLevel;
+    }>) => {
+      changeEventCounter++;
+      currentChangeEvent = { level: e.level.content, lastLevel: e.lastLevel.content };
+    };
+
+    editor.on('change', onChange);
+
+    assert.equal(changeEventCounter, 0, '0 event should be detected at start');
+    editor.undoManager.dispatchIfChanged();
+    assert.equal(changeEventCounter, 0, '0 event should be detected if content is coherent with the undoManager history');
+
+    Arr.last(editor.undoManager.data).each((lastLevel) => {
+      lastLevel.content = manualModifiedLevel;
+    });
+
+    assert.equal(editor.isDirty(), false, 'Editor should not be dirty before dispatchIfChanged');
+    editor.undoManager.dispatchIfChanged();
+    assertChangeEvent(currentChangeEvent);
+    assert.equal(changeEventCounter, 1, '1 event should be detected if the content is not coherent with the undoManager history');
+    assert.equal(editor.isDirty(), true, 'Editor should be dirty after dispatchIfChanged');
+
+    editor.undoManager.dispatchIfChanged();
+    assertChangeEvent(currentChangeEvent);
+
+    editor.undoManager.dispatchIfChanged();
+    assertChangeEvent(currentChangeEvent);
+
+    editor.undoManager.dispatchIfChanged();
+    assertChangeEvent(currentChangeEvent);
```

Comment:
```suggestion
    Arr.range(3, () => {
      editor.undoManager.dispatchIfChanged();
      assertChangeEvent(currentChangeEvent);
    });
```
No need to manually repeat this 3 times as we can use a loop to do that. This also helps to make it clearer about how many times it has run.

Reply:
done